### PR TITLE
aws: Add iam.GetRole

### PR DESCRIPTION
**Pull request by tolujimoh**
### Description

This PR adds functionality for calling iam.GetRole to support https://github.com/lyft/clutch-private/pull/2500

### CODE REVIEW

**Comment by mikecutalo** [comment]
> /ptal #core-datastores-prs

ptal commands dont work on oss repos btw

File: backend/mock/service/awsmock/awsmock.go [hunk 1]
```diff
@@ -37,6 +37,15 @@ func (s *svc) S3GetAccessPointPolicy(ctx context.Context, account, region, acces
 	}, nil
 }
 
+func (s *svc) GetRole(ctx context.Context, account, region, roleName string) (*iam.GetRoleOutput, error) {
```
Comment (mikecutalo) [inline backend/mock/service/awsmock/awsmock.go:40]: nit: maybe rename `GetIAMRole`?

Comment (tolujimoh) [reply backend/mock/service/awsmock/awsmock.go:40]: Sure. Can Update

File: backend/service/aws/iface.go [hunk 2]
```diff
@@ -31,6 +31,7 @@ type stsClient interface {
 
 type iamClient interface {
 	SimulateCustomPolicy(ctx context.Context, params *iam.SimulateCustomPolicyInput, optFns ...func(*iam.Options)) (*iam.SimulateCustomPolicyOutput, error)
+	GetRole(ctx context.Context, params *iam.GetRoleInput, optFns ...func(options *iam.Options)) (*iam.GetRoleOutput, error)
```
[the remark this answers is not in the document]
Comment (tolujimoh) [reply backend/service/aws/iface.go:34]: Thats the name of the AWS api
https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/iam#Client.GetRole

File: backend/service/aws/iam_test.go [hunk 3]
```diff
@@ -104,3 +106,10 @@ func (m *mockIAM) SimulateCustomPolicy(ctx context.Context, params *iam.Simulate
 	}
 	return m.getSimulationResults, nil
 }
+
+func (m *mockIAM) GetRole(ctx context.Context, params *iam.GetRoleInput, optFns ...func(*iam.Options)) (*iam.GetRoleOutput, error) {
```
[the remark this answers is not in the document]
Comment (tolujimoh) [reply backend/service/aws/iam_test.go:110]: Same with above

[the remark this answers is not in the document]
Comment (mikecutalo) [reply backend/service/aws/iam_test.go:110]: ah ok thank you

**Review by mikecutalo — APPROVED** [review]

**Pull request merged**